### ISSUE TITLE
Plotting FunGuild by group

### ISSUE THREAD

**Issue by alisr7** [ChiLiubio/microeco#46]
Hi Chi,

I'd like to plot FunGuild results by Group rather than Modules. Is there a way to do that I appreciate your help.
Thanks,
Ali

**Comment by ChiLiubio** [comment]
Hi, @alisr7 

Please see the parameter use_community in function cal_spe_func_perc. Then you can get the sample result for the following plotting.

Chi

**Comment by alisr7** [comment]
Hi Chi,

I was able to get it by sample, but I am trying to plot it by group mean. Is this possible?
I appreciate your help,
Ali

**Comment by ChiLiubio** [comment]
Hi,
Sure. For example, you can extract the result, i.e. a data frame, and put it in the tax_abund in your another cloned microtable object. Then you can plot them as you want, just like a taxa, although it is not an actually taxa. Thus, you can plot the barplot, heatmap ...  

Best
Chi

**Comment by alisr7** [comment]
Thanks Chi! That worked well.
Best,
Ali

**Comment by JBL05** [comment]
> Thanks Chi! That worked well. Best, Ali

Hi，Ali

I am new in R and microeco, I have the same issue, could you please share how you did that? Thank you very much.

Best,
JB

**Comment by ChiLiubio** [comment]
Hi @JBL05 

The easiest way is to use merge_sample function of microtable class to get a merged dataset, i.e. your group-named sample microtable object. Then the following operation is same with other steps.

Chi

**Comment by JBL05** [comment]
> Hi @JBL05
> 
> The easiest way is to use merge_sample function of microtable class to get a merged dataset, i.e. your group-named sample microtable object. Then the following operation is same with other steps.
> 
> Chi

Hi Prof. Chi,

Thank you very much for your quick reply.  I was trying to plot the result using a stacked barplot rather than modules. I created a new microtable class after I got object$res_spe_fun_perc following your suggestions by using microtable$new(), I created a new otu_table from object$res_spe_fun_perc, and a new tax_table by the predicted function guild names extract from object$res_spe_fun_perc rownames, the sample_table keep the same with the original dateset and it finally works. I hope I did that correctly.

Best,
JB
